xbmc/settings/dialogs/GUIDialogSettingsBase.cpp

```diff
@@ -193,9 +193,15 @@ bool CGUIDialogSettingsBase::OnMessage(CGUIMessage& message)
       int iControl = message.GetSenderId();
       if (iControl == CONTROL_SETTINGS_OKAY_BUTTON)
       {
-        OnOkay();
-        Close();
-        return true;
+        if (OnOkay())
+        {
+          Close();
+          return true;
+        }
+        else
+        {
+          return false;
+        }
```

Comment:
Maybe write
```cpp
if (!OnOkay())
  return false;

Close();
return true;
```

Reply:
Or to be consistent with the `CGUIDialogSettingsManagerBase::OnOkay()`:
```cpp
if (OnOkay())
{
  Close();
  return true;
}

return false;
```
Either way the style should be consistent :-)